server/src/lib/challenge.ts

```diff
@@ -31,74 +28,82 @@ export default class Challenge {
   }
 
   getPoints = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getPoints() - client_id:${client_id}, challenge_token:${challeng_token}`
-    );
-    let data = getMockData();
-
-    response.json(data.challengePoint);
+    response.json(await this.model.db.getUserPoints(client_id, challenge));
   };
 
   getWeeklyProgress = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getWeeklyProgress - client_id:${client_id}, challenge_token:${challeng_token}`
+    const progress = await this.model.db.getWeeklyProgress(
+      client_id,
+      challenge
     );
-    let data = getMockData();
-
-    response.json(data.weeklyChallenge);
+    const weeklyProgress = {
+      week: progress.week || null,
+      user: {
+        speak: progress.clip_count || null,
+        speak_total: progress.week == 2 ? 100 : 200,
+        listen: progress.vote_count || null,
+        listen_total: progress.week == 2 ? 50 : 100,
+      },
+      team: { invite: progress.colleague_count || null, invite_total: 50 },
+    };
+    response.json(weeklyProgress);
   };
 
   getTopMembers = async (
-    { client_id, params: { challeng_token, locale, type } }: Request,
+    { client_id, params: { challenge, locale, type }, query }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getTopMembers - client_id:${client_id}, locale: ${locale}, challenge_token:${challeng_token}, type:${type}`
+    const atype = type == 'vote' ? 'vote' : 'clip';
```

Comment:
I'm not familiar with the potential values of `type` – will it ever be something other than `"clip"` or `"vote"`?

Reply:
it is the type checking of typescript, I have to pass in literal clip or vote, I plan to make the parameters shorter and squeeze the following function call in a single line, but VS Code always re-format it in several lines. Maybe I should put the ternary expression in the function call.